md-dao/src/main/java/com/wipro/ats/bdre/md/dao/ProcessDependencyDAO.java

```diff
@@ -119,14 +120,14 @@
                     LOGGER.info("No downstream processes present");
                 }
                 //Listing upstream processes
-                Criteria checkUpstreamParentProcesses = session.createCriteria(Process.class).add(Restrictions.eq("process", passedProcess)).add(Restrictions.eq("deleteFlag", false)).setProjection(Projections.property("enqueuingProcessId"));
+                Criteria checkUpstreamParentProcesses = session.createCriteria(Process.class).add(Restrictions.eq("process", passedProcess)).add(Restrictions.eq(DELETE_FLAG, false)).setProjection(Projections.property("enqueuingProcessId"));
                 List<Integer> upstreamParentProcessList = new ArrayList<Integer>();
-                if (checkUpstreamParentProcesses.list().size() != 0) {
+                if (!checkUpstreamParentProcesses.list().isEmpty()) {
                     upstreamParentProcessList = checkUpstreamParentProcesses.list();
 
-                    Criteria checkUpstreamProcesses = session.createCriteria(Process.class).add(Restrictions.in("processId", upstreamParentProcessList)).add(Restrictions.eq("deleteFlag", false));
+                    Criteria checkUpstreamProcesses = session.createCriteria(Process.class).add(Restrictions.in("processId", upstreamParentProcessList)).add(Restrictions.eq(DELETE_FLAG, false));
                     List<Process> upstreamProcessList = new ArrayList<Process>();
-                    if (checkUpstreamProcesses.list().size() != 0) {
+                    if (!checkUpstreamProcesses.list().isEmpty()) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AS134)